docs/datamaps/index.md

```diff
@@ -0,0 +1,164 @@
+You can query a data map value using `Holder#getData(DataMapType)`. If that object doesn't have a value attached, the method will return `null`.
+
+:::note
+Only reference holders will return a value in that method. `Named` holders will **not**.  Generally, you will only encounter reference holders (which are returned by methods such as `Registry#wrapAsHolder`, `Registry#getHolder` or the different `builtInRegistryHolder` methods).
+:::
+
+To continue the example above, we can implement our intended behaviour as follows:
+```java
+public static void onItemDrop(final ItemTossEvent event) {
+    final ItemStack stack = event.getEntity().getItem();
+    // ItemStack has a getItemHolder method that will return a Holder<Item> which points to the item the stack is of
+    //highlight-next-line
+    final DropHealing value = stack.getItemHolder().getData(DROP_HEALING);
+    // Since getData returns null if the item will not have a drop healing value attached, we guard against it being null
+    if (value != null) {
+        // And here we simply use the values
+        if (event.getPlayer().level().getRandom().nextFloat() > value.chance()) {
+            event.getPlayer().heal(value.amount());
+        }
+    }
+}
+```
+
+## Advanced data maps
+Advanced data maps are data maps which have added functionality. Namely, the ability of merging values and selectively removing them, through a remover. Implementing some form of merging and removers is highly recommended for data maps whose values are collection-likes (like `Map`s or `List`s).
+
+`AdvancedDataMapType` have one more generic besides `T` and `R`: `VR extends DataMapValueRemover<T, R>`. This additional generic allows you to datagen remove objects with increased type safety, but it can otherwise be ignored and treated as wilcard (`?`) for most use cases.
+
+### Creation
+You create an `AdvancedDataMapType` using `AdvancedDataMapType#builder`. Unlike the normal builder, the builder returned by that method will have two more methods (`merger` and `remover`), and it will return an `AdvancedDataMapType`.  
+
+Registration methods remain the same.
+
+### Mergers
+An advanced data map can provide a `DataMapValueMerger` through `AdvancedDataMapType#merger`. This merger will be used to handle conflicts between data packs that attempt to attach a value to the same object.  
+The merger will be given the two conflicting values, and their sources (as an `Either<TagKey<R>, ResourceKey<R>>` since values can be attached to all entries within a tag, not just individual entries), and is expected to return the value that will actually be attached.  
+Generally, mergers should simply merge the values, and should not perform "hard" overwrites unless necessary (i.e. if merging isn't possible). A value can be overwritten by specifying the object-level `replace` field, which will bypass the merger.
```

Comment:
Hmm? There are use cases for bypassing mergers, especially for collections. You may want to overwrite just a value because another pack adds an entry you don't like, while continuing to merge the others.